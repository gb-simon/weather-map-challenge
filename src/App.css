.App {
  text-align: center;
  color: white;
}

.secondaryTitle {
  margin-left: 35px !important;
}

.box {
  max-height: 200px;
  max-width: 200px;
  background-color: #b20000;
}
