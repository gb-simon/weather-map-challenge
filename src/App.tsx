import React, { useEffect, useState } from "react";
import "./App.css";
import {
  Box,
  Grid,
  MenuItem,
  Paper,
  TextField,
  Typography,
  styled,
} from "@mui/material";
import { cities, myWeather } from "./Objects";

function App() {
  const [weatherInfo, setWeatherInfo] = useState<myWeather>();
  const [location, setLocation] = useState<string>(cities[1].label);
  const [reload, setReload] = useState<boolean>(false);
  const URL = `https://api.openweathermap.org/data/2.5/weather?q=${location}&APPID=4f614993e61b575abedbf67c0b5a4514`;

  //Since the API doesn't return Celcius I have to convert from Kelvin
  //(( kelvinValue - 273.15) * 9/5) + 32

  const Item = styled(Paper)(({ theme }) => ({
    ...theme.typography.body2,
    textAlign: "center",
    backgroundColor: "#b20000", //    backgroundColor: cities.location.color,
    color: theme.palette.text.secondary,
    width: 100,
    height: 100,
    fontSize: "16px",
    lineHeight: "100px",
  }));

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setReload(!reload);
    setLocation(event.target.value);
  };

  useEffect(() => {
    fetch(URL)
      .then((response) => response.json())
      .then((data) => setWeatherInfo(data))
      .catch((err) => console.error(err));
  }, [reload]);

  console.log(weatherInfo);

  return (
    <Grid
      container
      justifyContent="center"
      alignItems="center"
      direction={{ xs: "row", sm: "column" }}
      p={2}
      className="App"
    >
      <Paper
        elevation={1}
        sx={{
          backgroundColor: "#FF4C4C",
          maxWidth: "600px",
        }}
      >
        <Grid
          container
          direction={{ xs: "row", sm: "column" }}
          p={2}
          className="App"
        >
          <Grid container item justifyContent="center">
            <TextField
              select
              label="City"
              value={location}
              onChange={handleChange}
              helperText="Please select a city"
            >
              {cities.map((option) => (
                <MenuItem key={option.value} value={option.value}>
                  {option.label}
                </MenuItem>
              ))}
            </TextField>
          </Grid>
          <Grid container item justifyContent="center" alignItems="center">
            <Typography variant="h1">{weatherInfo?.main.temp}</Typography>
            <Typography className="secondaryTitle" variant="h3">
              {weatherInfo?.weather[0].description}
            </Typography>
          </Grid>
          <Grid container item justifyContent="center">
            <Box
              sx={{
                p: 2,
                display: "flex",
                gap: 2,
              }}
            >
              <Item elevation={3}>
                {`Feels ${weatherInfo?.main.feels_like}`}
              </Item>
              <Item elevation={3}>
                {`Humidity ${weatherInfo?.main.humidity}`}
              </Item>
              <Item elevation={3}>{`Max ${weatherInfo?.main.temp_max}`}</Item>
              <Item elevation={3}>{`Min ${weatherInfo?.main.temp_min}`}</Item>
            </Box>
          </Grid>
        </Grid>
      </Paper>
    </Grid>
  );
}

export default App;
