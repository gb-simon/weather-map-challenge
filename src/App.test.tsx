import React from "react";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import App from "../src/App";
import {
  fireEvent,
  getByRole,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import UserEvent from "@testing-library/user-event";
import { MenuItem, Select } from "@mui/material";

test("the selector renders the city the user picked", async () => {
  render(<App />);
  await UserEvent.click(getByRole(screen.getByTestId("city"), "button"));
  waitFor(() => UserEvent.click(screen.getByText("Paris")));
  expect(screen.getByText("Paris"));
});

describe("App home page", () => {
  it("renders the home page", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("should correctly set default option", () => {
    render(<App />);
    const header = screen.getByText("Buenos Aires");
    expect(header).toBeInTheDocument();
  });

  it("renders a selector that shows five city options", () => {
    const { getByRole } = render(
      <>
        <Select defaultValue="Buenos Aires">
          <MenuItem value="Buenos Aires">Buenos Aires</MenuItem>
          <MenuItem value="Cordoba">Córdoba</MenuItem>
          <MenuItem value="Quebec">Québec</MenuItem>
          <MenuItem value="Paris">Paris</MenuItem>
          <MenuItem value="Beijing">Beijing</MenuItem>
        </Select>
      </>
    );

    fireEvent.mouseDown(getByRole("button"));
    expect(getByRole("listbox")).toHaveTextContent("Paris");
    expect(getByRole("listbox")).toHaveTextContent("Québec");
    expect(getByRole("listbox")).toHaveTextContent("Buenos Aires");
    expect(getByRole("listbox")).toHaveTextContent("Córdoba");
  });
});
