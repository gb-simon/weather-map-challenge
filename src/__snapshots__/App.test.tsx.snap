// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App home page renders the home page 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container app css-bydcm4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1hfdk1f-MuiGrid-root"
    >
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary selector css-qfxojq-MuiInputBase-root-MuiFilledInput-root-MuiSelect-root"
        data-testid="city"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="city select"
          class="MuiSelect-select MuiSelect-filled MuiInputBase-input MuiFilledInput-input css-d9oaum-MuiSelect-select-MuiInputBase-input-MuiFilledInput-input"
          id="select"
          role="button"
          tabindex="0"
        >
          Buenos Aires
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="Buenos Aires"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconFilled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-[object Object] css-8oybt5-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-6 MuiGrid-direction-xs-column MuiGrid-grid-xs-4 css-1ix7mb5-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <strong
              class="locationTitle"
            >
              , 
            </strong>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <strong
              class="dateTitle"
            >
              Aug 28, 2022
            </strong>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-1p27619-MuiGrid-root"
          >
            <strong
              class="temp"
            >
              0°
            </strong>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
          >
            <strong
              class="descriptions"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-1 css-gc9e0t-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-[object Object] info css-h7y787-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-1l5akr4"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="OpacityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.66 8 12 2.35 6.34 8C4.78 9.56 4 11.64 4 13.64s.78 4.11 2.34 5.67 3.61 2.35 5.66 2.35 4.1-.79 5.66-2.35S20 15.64 20 13.64 19.22 9.56 17.66 8zM6 14c.01-2 .62-3.27 1.76-4.4L12 5.27l4.24 4.38C17.38 10.77 17.99 12 18 14H6z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Humidity %undefined
              </p>
            </div>
            <div
              class="MuiBox-root css-1l5akr4"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="AddCircleOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Max 0 °
              </p>
            </div>
            <div
              class="MuiBox-root css-1l5akr4"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="RemoveCircleOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Min 0°
              </p>
            </div>
            <div
              class="MuiBox-root css-1l5akr4"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="WindPowerIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 3h6v2H4zM1 7h5v2H1zm2 12h5v2H3zm10.73-8.39c.75.23 1.3.78 1.57 1.46l4.27-7.11c.65-1.08.3-2.48-.78-3.13-.87-.52-1.99-.41-2.73.29l-3.43 3.21c-.4.37-.63.9-.63 1.45v3.93c.36-.15.98-.33 1.73-.1zm-3.12 1.66c.16-.52.48-.96.89-1.27H3.28C2.02 11 1 12.02 1 13.28c0 1.02.67 1.91 1.65 2.19l4.51 1.29c.53.15 1.1.08 1.58-.21l2.69-1.61c-.77-.62-1.13-1.67-.82-2.67zm11.6 6.34-2.28-4.1c-.27-.48-.73-.83-1.26-.97l-3.18-.8c.03.32 0 .66-.1.99-.32 1.06-1.28 1.77-2.39 1.77-.61 0-.99-.22-1-.22V21c-1.1 0-2 .9-2 2h6c0-1.1-.9-2-2-2v-4.28l4.61 4.61c.89.89 2.33.89 3.22 0 .72-.72.88-1.83.38-2.72z"
                />
                <path
                  d="M12.56 14.43c.79.24 1.63-.2 1.87-1 .24-.79-.2-1.63-1-1.87-.79-.24-1.63.2-1.87 1-.24.79.21 1.63 1 1.87z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Wind undefined
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
